internal/hcp/registry/json.go

```diff
@@ -93,7 +93,19 @@ func (h *JSONRegistry) CompleteBuild(
 	artifacts []sdkpacker.Artifact,
 	buildErr error,
 ) ([]sdkpacker.Artifact, error) {
-	return h.bucket.completeBuild(ctx, build.Name(), artifacts, buildErr)
+	name := build.Name()
+	pluginMetadata := map[string]packer.PluginDetails{}
+	coreBuild, ok := build.(*packer.CoreBuild)
```

Comment:
You probably don't need to check for correctness here, it's already a `packer.CoreBuild` in effect, although typed by an interface.

We should definitely follow-up this PR and change this typing to use the concrete type instead of the interface since it's virtually (pun intended) useless to type it this way, let's simplify our lives here.